Userland/Libraries/LibWeb/SVG/SVGPathElement.cpp

```diff
@@ -390,8 +390,33 @@ float PathDataParser::parse_fractional_constant()
 float PathDataParser::parse_number()
 {
     auto number = parse_fractional_constant();
-    if (match('e') || match('E'))
-        TODO();
+
+    if (!match('e') && !match('E'))
+        return number;
+    consume();
+
+    auto exponent_sign = parse_sign();
+
+    StringBuilder exponent_builder;
+    while (!done() && isdigit(ch()))
+        exponent_builder.append(consume());
+    VERIFY(exponent_builder.length() > 0);
+
+    auto exponent = exponent_builder.to_string().to_int().value();
+
+    if (number == 0) // fast path: if the number is 0 theres no point to computing the exponentiation
```

Comment:
theres -> there's

Reply:
In fact, please write comments as proper sentences, e.g

```cpp
// Fast path: If the number is 0, there's no point in computing the exponentiation.
```

Reply:
sorry, fixed!